### PR TITLE
Use one EventLoopGroup in all http2NetworkClients

### PR DESCRIPTION
**Pull request by zzmao**
Use one EventLoopGroup in all http2NetworkClients. This is suggested in Netty best practice: The change is based on a best practice of netty: http://normanmaurer.me/presentations/2014-facebook-eng-netty/slides.html#25.0

### CODE REVIEW

**Comment by justinlin-linkedin** [comment]
I don't really understand this PR. One OperationController in router would create one HTTP2NetworkClient, which would have its own eventloop. But with this change, we are forcing all the Http2NetworkClient in all OperationController to use the same eventloop?

**Comment by justinlin-linkedin** [comment]
And did you see any performance boost with this change? And can you explain the reason why if there is any? I would be surprised if there is any performance benefit coming from this change.

**Comment by zzmao** [comment]
@justinlin-linkedin  The change is based on a best practice of netty: http://normanmaurer.me/presentations/2014-facebook-eng-netty/slides.html#25.0

It's explainable, because a set of eventloopgroup has # of CPU threads. If more operation controllers, it would be multiple sets of threads. More threads cause more context switch. I think that's why the doc suggest to reuse eventloopgroup.

**Comment by justinlin-linkedin** [comment]
> @justinlin-linkedin The change is based on a best practice of netty: http://normanmaurer.me/presentations/2014-facebook-eng-netty/slides.html#25.0
> 
> It's explainable, because a set of eventloopgroup has # of CPU threads. If more operation controllers, it would be multiple sets of threads. More threads cause more context switch. I think that's why the doc suggest to reuse eventloopgroup.

Can you add this to description so we will have a better commit message for this PR.

File: ambry-server/src/integration-test/java/com/github/ambry/server/Http2NetworkClientTest.java [hunk 1]
```diff
@@ -97,6 +102,8 @@ public static void initializeTests() throws Exception {
         "http2-client");
     TestSSLUtils.addHttp2Properties(clientSSLProps, SSLFactory.Mode.CLIENT, false);
     clientSSLConfig = new SSLConfig(new VerifiableProperties(clientSSLProps));
+    eventLoopGroup = new NioEventLoopGroup();
```
Comment (justinlin-linkedin) [inline ambry-server/src/integration-test/java/com/github/ambry/server/Http2NetworkClientTest.java:105]: nit: this line is no use at all.

**Review by justinlin-linkedin — APPROVED** [review]

**Pull request merged**